src/test/fuzz/txrequest.cpp

```diff
@@ -0,0 +1,374 @@
+            m_announcements[txhash][peer].m_time = exptime;
+        }
+
+        // Add event so that AdvanceToEvent can quickly jump to the point where its exptime passes.
+        if (exptime > m_now) m_events.push(exptime);
+
+        // Call TxRequestTracker's implementation.
+        m_tracker.RequestedTx(peer, TXHASHES[txhash], exptime);
+    }
+
+    void ReceivedResponse(int peer, int txhash)
+    {
+        // Apply to naive structure: convert anything to COMPLETED.
+        if (m_announcements[txhash][peer].m_state != State::NOTHING) {
+            m_announcements[txhash][peer].m_state = State::COMPLETED;
+            Cleanup(txhash);
+        }
+
+        // Call TxRequestTracker's implementation.
+        m_tracker.ReceivedResponse(peer, TXHASHES[txhash]);
+    }
+
+    void GetRequestable(int peer)
+    {
+        // Implement using naive structure:
+
+        //! list of (sequence number, txhash, is_wtxid) tuples.
+        std::vector<std::tuple<uint64_t, int, bool>> result;
+        std::vector<std::pair<NodeId, GenTxid>> expected_expired;
+        for (int txhash = 0; txhash < MAX_TXHASHES; ++txhash) {
+            // Mark any expired REQUESTED announcements as COMPLETED.
+            for (int peer2 = 0; peer2 < MAX_PEERS; ++peer2) {
+                Announcement& ann2 = m_announcements[txhash][peer2];
+                if (ann2.m_state == State::REQUESTED && ann2.m_time <= m_now) {
+                    expected_expired.emplace_back(peer2, GenTxid{ann2.m_is_wtxid, TXHASHES[txhash]});
+                    ann2.m_state = State::COMPLETED;
+                    break;
+                }
+            }
+            // And delete txids with only COMPLETED announcements left.
+            Cleanup(txhash);
+            // CANDIDATEs for which this announcement has the highest priority get returned.
+            const Announcement& ann = m_announcements[txhash][peer];
+            if (ann.m_state == State::CANDIDATE && GetSelected(txhash) == peer) {
+                result.emplace_back(ann.m_sequence, txhash, ann.m_is_wtxid);
+            }
+        }
+        // Sort the results by sequence number.
+        std::sort(result.begin(), result.end());
+        std::sort(expected_expired.begin(), expected_expired.end());
+
+        // Compare with TxRequestTracker's implementation.
+        std::vector<std::pair<NodeId, GenTxid>> expired;
+        const auto actual = m_tracker.GetRequestable(peer, m_now, &expired);
+        std::sort(expired.begin(), expired.end());
+        assert(expired == expected_expired);
+
+        m_tracker.PostGetRequestableSanityCheck(m_now);
+        assert(result.size() == actual.size());
+        for (size_t pos = 0; pos < actual.size(); ++pos) {
+            assert(TXHASHES[std::get<1>(result[pos])] == actual[pos].GetHash());
+            assert(std::get<2>(result[pos]) == actual[pos].IsWtxid());
+        }
+    }
+
+    void Check()
+    {
+        // Compare CountTracked and CountLoad with naive structure.
```

Comment:
You've regressed this comment in the latest force push. It should say:

`// Compare Size(), Count(), CountInFlight(), and CountCandidates() with naive structure.`